### PR TITLE
componentGroup migration fix: don't apply for non-fragments

### PR DESCRIPTION
**Pull request by arkadius**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Review by lukasz-bigorajski — APPROVED** [review]
LGTM

File: engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/join/TableJoinComponent.scala [hunk 1]
```diff
@@ -264,6 +264,7 @@ object TableJoinComponent
   }
 
   override def typesToExtract: List[TypedClass] = List(
+    Typed.typedClass[BranchType],
```
Comment (wrzontek) [inline engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/join/TableJoinComponent.scala:267]: what is this sneaky little line? :D

Comment (wrzontek) [reply engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/join/TableJoinComponent.scala:267]: LGTM aside from this

Comment (arkadius) [reply engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/join/TableJoinComponent.scala]: removed

**Pull request merged**